Mathlib/Data/PFunctor/Univariate/M.lean

```diff
@@ -790,14 +790,13 @@ set_option linter.uppercaseLean3 false in
 
 /-- corecursor where it is possible to return a fully formed value at any point
 of the computation -/
-def corec' {α : Type u} (F : ∀ {X : Type u}, (α → X) → α → Sum (M P) (P X)) (x : α) : M P :=
-  corec₁
-    (fun _ rec (a : Sum (M P) α) =>
-      let y := a >>= F (rec ∘ Sum.inr)
-      match y with
-      | Sum.inr y => y
-      | Sum.inl y => (rec ∘ Sum.inl) <$> M.dest y)
-    (@Sum.inr (M P) _ x)
+def corec' (F : α → M P ⊕ P α) (x : α) : M P :=
+  M.corec
+    (fun (a : M P ⊕ α) =>
+      match Sum.bind a F with
+      | Sum.inr y => P.map Sum.inr y
+      | Sum.inl y => P.map Sum.inl (M.dest y))
+    (Sum.inr x)
```

Comment:
This looks like a pretty different statement to me; did you mean to change this in this PR, or was it for a future PR? If it really is supposed to be here, please describe it in the PR description,